appinventor/components/src/com/google/appinventor/components/runtime/LocationSensor.java

```diff
@@ -84,8 +84,13 @@ public void onLocationChanged(Location location) {
         hasAltitude = true;
         altitude = location.getAltitude();
       }
-      hasLocationData = true;
-      LocationChanged(latitude, longitude, altitude);
+
+      // By default Location.latitude == Location.longitude == 0.
+      // So we want to ignore that case rather than generating a changed event.
+      if (longitude != UNKNOWN_VALUE && latitude != UNKNOWN_VALUE) {
```

Comment:
should this be OR, not AND?
what happens if the latitude is 0 and the longitude is non-zero?